lib/modules/manager/npm/post-update/pnpm.ts

```diff
@@ -70,7 +75,14 @@ export async function generateLockFile(
       extraEnv.NPM_EMAIL = env.NPM_EMAIL;
     }
 
-    let args = '--recursive --lockfile-only';
+    let args = '--lockfile-only';
+    if (
+      await localPathExists(
+        getSiblingFileName(lockFileName, 'pnpm-workspace.yaml'),
+      )
+    ) {
+      args += ' --recursive';
```

Comment:
I wonder if we need this flag at all. 🤔

at least pnpm v9+ is doing recursive install automatically when it detects workspaces.

Reply:
that will be a breaking change for pnpm@8 I guess?

Reply:
That's hard to implement in semver and too many pnpm source, I don't very want to do it 😂

Reply:
what? isn't it simply if not intersects with `>=9`?

Reply:
Ok I don't know there is a semver.intersects, I was looking on versioning/semver and I thought  we only have that on versioning/composer.